ls1intum/Artemis
# `Quiz exercises`: Fix visual selection when multiple short answer question exists

### Checklist
#### General


#### Client

### Motivation and Context



Adding spot through visual mode does not work if there are multiple short answer questions in 1 quiz exercise.

https://user-images.githubusercontent.com/10885503/209796414-f905d4d0-8adb-40c8-a406-44957787ae9d.mov

### Description


The id of the short answer questions is the same for all questions. The element that is referenced by the id is used to determine which question is being selected when adding a spot through visual mode. Changing the id to use angular reference solves the issue.

### Steps for Testing

Prerequisites:
- 1 Instructor
- 1 Course
- 1 Quiz Exercise with multiple short answer questions

1. Log in to Artemis
2. Navigate to Course Administration
3. Select the Course
4. Navigate to Exercises
5. Select the Exercise
6. Click Visual on the first short answer question
7. Highlight the word and click Add Spot. (Verify that it works)
8. Click Visual on the second short answer question
9. Highlight the word and click Add Spot. (Verify that it still works)

### Review Progress






#### Code Review
#### Manual Tests

### Test Coverage



| Class/File | Branch | Line |
|------------|-------:|-----:|
| short-answer-question-edit.component.ts | 70.32% | 93.81% |

### Screenshots



https://user-images.githubusercontent.com/10885503/209792501-392dee24-41a9-4526-89e4-24dbcdd55429.mov
